oshdb-util/src/main/java/org/heigit/ohsome/oshdb/util/time/TimestampFormatter.java

```diff
@@ -11,28 +11,8 @@
  */
 public class TimestampFormatter {
 
-  private static TimestampFormatter instance;
-  private final ThreadLocal<SimpleDateFormat> formatDate = ThreadLocal.withInitial(() -> {
-    SimpleDateFormat ret = new SimpleDateFormat("yyyy-MM-dd");
-    ret.setTimeZone(TimeZone.getTimeZone("UTC"));
-    return ret;
-  });
-  private final ThreadLocal<SimpleDateFormat> formatIsoDateTime = ThreadLocal.withInitial(() -> {
-    SimpleDateFormat ret = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
-    ret.setTimeZone(TimeZone.getTimeZone("UTC"));
-    return ret;
-  });
-
-  /**
-   * Get a standard TimestampFormatter.
-   *
-   * @return a timestamp formatter object
-   */
-  public static TimestampFormatter getInstance() {
-    if (instance == null) {
-      instance = new TimestampFormatter();
-    }
-    return instance;
+  private TimestampFormatter() {
+    // utility class
   }
```

Comment:
the background here was/is, that SimpleDateFormat is not thread safe, so using one per thread solves this. But TimestampFormatter was/is a singleton so removing those thread-local was not possible, 
I would like to see/know if this current solution has any negative impacted in performance, if not I think this is much cleaner.
Let's see the benchmarks for this.

Reply:
an other solution would be to make TimestampFormater a singleton again but use synchonized-blocks to ensure, that the Dateformater is not concurrently used.
https://docs.oracle.com/javase/tutorial/essential/concurrency/locksync.html

Reply:
1. AFAICS, the class is actually not really used in the OSHDB itself (apart from a single usage in `MapReducer.java` which would be "trivial" to reimplement). It is however used quite extensively by the ohsome API in data extraction requests.
2. For the usage in the ohsome API, the current implementation with threadlocal `SimpleDateFormat`ers is actually quite crucial for performance: we don't want to re-parse the data format pattern every time we want to stringify a datetime value. The singleton+threadlocal pattern was the easiest I could come up with.
3. So I guess we should either a) remove the class here and put the more complex but better performing logic into the ohsome API, or b) keep it as is.

does that make sense?

Reply:
> 2\. For the usage in the ohsome API, the current implementation with threadlocal `SimpleDateFormat`ers is actually quite crucial for performance: we don't want to re-parse the data format pattern every time we want to stringify a datetime value. The singleton+threadlocal pattern was the easiest I could come up with.

Do  you have any numbers on here, would be interesting to know how negative the affect of creating new DateFormater actually is.

Reply:
> 1. AFAICS, the class is actually not really used in the OSHDB itself (apart from a single usage in `MapReducer.java` which would be "trivial" to reimplement). It is however used quite extensively by the ohsome API in data extraction requests.

I looked through the reference to `TimestampFormatter` in the ohsome-api, and as far as I can tell only in the 
- [DataExtractionTransformer](https://github.com/GIScience/ohsome-api/blob/0965541e7a24171cd1d11017eed7fc781b29b851/src/main/lombok/org/heigit/ohsome/ohsomeapi/executor/DataExtractionTransformer.java#L144)
- [ElementsRequestExecutor](https://github.com/GIScience/ohsome-api/blob/0965541e7a24171cd1d11017eed7fc781b29b851/src/main/lombok/org/heigit/ohsome/ohsomeapi/executor/ElementsRequestExecutor.java#L121)
the LocalThread mechanism is used. And also only for `isoDateTime`.

So my preference is than to remove the TimestampFormatter from the oshdb-api and move it to the ohsome-api maybe in a slightly modified version. For example a TimestampFormatter per request with proper clean up afterwards. this should then not influence performance.

What are your  thoughts about this? And should we already do it in this PR, if not, I will revert this change and we should create an issue.

Reply:
> Do you have any numbers on here, would be interesting to know how negative the affect of creating new DateFormater actually is.

I created a simple [microbenchmark for this](https://gitlab.gistools.geog.uni-heidelberg.de/giscience/big-data/ohsome/microbenchmarks/timestampformatter-benchmark). These are the results:

```
Benchmark                          Mode  Cnt  Score   Error  Units
TimestampFormatterBench.master    thrpt    8  2.753 ± 0.522  10⁶ ops/s
TimestampFormatterBench.pr419     thrpt    8  0.919 ± 0.211  10⁶ ops/s

Reply:
Thank you, and yes sure, there is a performance difference between creating always a new one. 
But I would be interested, if we can see this in the application/request. I guess this is more cumbersome to measure this.

Reply:
I think it's pretty much neglectable in the actual application. Yes, it might get called many times in a data extraction request, and the overhead is significant (~3x as seen above). But for data extraction requests, there should be other parts of the processing taking much longer than this (e.g. the building of the geojson output for example), and the real bottleneck will practically always be the network.

I had thought that the overhead of recreating the formatter objects would be much larger. So, given that that's not the case, I guess we could simplify the implementation here. But then the question remains: why do we need to ship this thin wrapper around an already built-in java functionality? Woudn't it be better drop the class altogether?

Reply:
Thank you and I totally agree with you. So dropping this class in the oshdb-api, replacing the functionality in the oshdb-api and find a new (or same) solution for the ohsome-api.
So I do the first 2 parts in the oshdb-api right now in this PR.

Reply:
Done and added to the changelog.